File: 3860-resulting-string-after-adjacent-removals/3860-resulting-string-after-adjacent-removals.py
class Solution:
    def resultingString(self, s: str) -> str:
        st = []

        for c in s:
            st.append(c)

            while len(st) > 1 and (
    (ord(st[-1]) - ord('a')) % 26 == (ord(st[-2]) - ord('a') + 1) % 26 or
    (ord(st[-2]) - ord('a')) % 26 == (ord(st[-1]) - ord('a') + 1) % 26
):
                st.pop()
                st.pop()

        return ''.join(st) 

File: 3860-resulting-string-after-adjacent-removals/3860-resulting-string-after-adjacent-removals.cpp
class Solution {
public:
    string resultingString(string s) {
        string st;

        for (char c : s) {
            st.push_back(c);

            while (st.size() > 1 &&
                  ((st.back() - 'a') % 26 == (st[st.size() - 2] - 'a' + 1) % 26 ||
                   (st[st.size() - 2] - 'a') % 26 == (st.back() - 'a' + 1) % 26)) {
                st.pop_back();  // Remove top
                st.pop_back();  // Remove second top
            }
        }

        return st;
    }
};